### ISSUE TITLE
[HOLD for payment 2023-04-03] [Hold - Upstream fix Plaid #512][$1000] DEV - console error Warning: `Internal React error: Attempted to capture` when clicking X button in "Connect to bank account" page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13917]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to Settings >  workspace > Connect bank account > Connect connect online with Plaid
2. click :x: button

## Expected Result:
should not show any console error

## Actual Result:
 shows console error on  click :x: button

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] iOS / native

**Version Number:** v1.2.45-0(main branch)
**Reproducible in staging?:**
**Reproducible in production?:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/210185700-426a130b-0534-474b-bedd-fd98d47ac678.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1672400428847699

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01090ed9dedf023458</li>
        <li>Upwork Job ID: 1615420052392824832</li>
        <li>Last Price Increase: 2023-02-02</li>
    </ul>
</details>

**Comment by michaelhaxhiu** [comment]
Hmm this is likely indicative of an underlying bug/error in the prod environment. I'm not able to reproduce this one as a non-eng, so I'm going to move to add an engineer internally and get their input.

**Comment by sobitneupane** [comment]
> Hmm this is likely indicative of an underlying bug/error in the prod environment. I'm not able to reproduce this one as a non-eng, so I'm going to move it forward to add an engineer and C+ for their input.

I was able to reproduce. Such Console Warnings appear only on Dev.

**Comment by michaelhaxhiu** [comment]
Perfect thanks sobit! That works as a proxy check 👍 and I think we are fixing these types of issues too

**Comment by michaelhaxhiu** [comment]
@luacmartins lmk if you think this is best handled `internally` or `externally`

**Comment by luacmartins** [comment]
We can go external with this

**Comment by s77rt** [comment]
### Proposal
https://github.com/plaid/react-native-plaid-link-sdk/pull/512

### RCA
The issue is in `react-native-plaid-link-sdk`
Please check the PR for more info

**Comment by luacmartins** [comment]
It seems like the fix is in the external lib. There's an open PR against it, but no movement there yet.

**Comment by thienlnam** [comment]
Upwork automation kinda breaks when we go from External -> Internal right now, I just reset it so it's basically created as a new External job

**Comment by luacmartins** [comment]
There's some movement in the [Plaid PR](https://github.com/plaid/react-native-plaid-link-sdk/pull/512)

**Comment by michaelhaxhiu** [comment]
Yep, changes were approved yesterday in the PR. Hoping things keep going forward! 

@s77rt did you apply for the[ job in Upwork](https://www.upwork.com/jobs/~01090ed9dedf023458) yet?

**Comment by s77rt** [comment]
@michaelhaxhiu No, I have not been assigned the issue yet.

**Comment by luacmartins** [comment]
The approval on the PR seems to be from an external contributor. Hopefully a library maintainer will get on it soon!

**Comment by luacmartins** [comment]
Still no movement on https://github.com/plaid/react-native-plaid-link-sdk/pull/512

**Comment by luacmartins** [comment]
Still no movement in the plaid PR

**Comment by luacmartins** [comment]
Still no movement in the plaid PR

**Comment by sobitneupane** [comment]
Still no movement in [the plaid PR](https://github.com/plaid/react-native-plaid-link-sdk/pull/512).

**Comment by s77rt** [comment]
@luacmartins @michaelhaxhiu @sobitneupane Shouldn't we contact the repo maintainer? Is the last price increase valid?

I think one of you need to review my proposal, if it's okay we can hold the issue in favor of the upstream PR to avoid further invalid increases.

**Comment by luacmartins** [comment]
Thanks for the ping @s77rt! Sorry, I should have assigned you to the issue before. @michaelhaxhiu can we revert the last price increase? We already had a solution in place, we are just waiting for the external maintainers to approve/merge.

**Comment by luacmartins** [comment]
Still no movement from Plaid

**Comment by luacmartins** [comment]
Moving to a weekly since we're waiting for an upstream fix.

**Comment by MelvinBot** [comment]
This issue has not been updated in over ***15*** days. @michaelhaxhiu, @s77rt, @luacmartins, @sobitneupane eroding to ***Monthly*** issue.

P.S. Is everyone reading this sure this is really a near-term priority? Be brave: if you disagree, go ahead and close it out. If someone disagrees, they'll reopen it, and if they don't: one less thing to do!

**Comment by s77rt** [comment]
Still waiting for PR to get merged upstream.

**Comment by s77rt** [comment]
The upstream PR has been merged :tada:

**Comment by s77rt** [comment]
Not sure when the fix will land though ^

**Comment by luacmartins** [comment]
Still waiting on a Plaid release with the fix.

**Comment by s77rt** [comment]
What a timing :D The new release is here :tada:  https://github.com/plaid/react-native-plaid-link-sdk/releases/tag/v10.0.0
Will work on the PR asap.

**Comment by s77rt** [comment]
Looks like we have another error unrelated to this issue that is present on `main`. Probably need to handle this in another issue.

![Screenshot from 2023-03-21 21-42-59](https://user-images.githubusercontent.com/16493223/226736101-fcd5bec2-195b-419b-8749-af7723b9d8e6.png)

**Comment by s77rt** [comment]
[PR](https://github.com/Expensify/App/pull/16361) is ready.

___

@luacmartins @michaelhaxhiu Can you please update:
1. Upwork job as current one is expired
2. The issue title (remove the hold)

**Comment by luacmartins** [comment]
Ah nice! I'll check the PR tomorrow! Thank you!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.89-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/16361


If no regressions arise, payment will be issued on 2023-04-03. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@s77rt / @sobitneupane / @luacmartins] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@s77rt / @sobitneupane / @luacmartins] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt / @sobitneupane / @luacmartins] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@michaelhaxhiu] Determine if we should create a regression test for this bug.
- [ ] [@s77rt / @sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@michaelhaxhiu] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: N/A This is a bug upstream
- The offending PR has been commented on: N/A
- A discussion in #expensify-bugs has been started: I think this does not apply here

___

**Regression Test Proposal**
1. Login to App
2. Go to Settings > Workspaces > any workspace > Connect bank account > Connect connect online with Plaid
3. Click X button
4. Verify that you don't get this error: `Internal React error: Attempted to capture a commit phase error inside a detached tree.`

Note: Step 4 would ideally be: "4. Verify that you don't get any error" however there are some other errors at the moment that are unrelated to this PR.

**Comment by MelvinBot** [comment]
@michaelhaxhiu, @s77rt, @luacmartins, @sobitneupane Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@michaelhaxhiu, @s77rt, @luacmartins, @sobitneupane Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by s77rt** [comment]
I think we are waiting for @michaelhaxhiu to complete the checklist

**Comment by s77rt** [comment]
@michaelhaxhiu @luacmartins Can you please help on processing the payment here? The listed Upwork job is expired

**Comment by MelvinBot** [comment]
@michaelhaxhiu, @s77rt, @luacmartins, @sobitneupane Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by luacmartins** [comment]
@michaelhaxhiu can you help with payment here?

**Comment by MelvinBot** [comment]
@michaelhaxhiu, @s77rt, @luacmartins, @sobitneupane Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by michaelhaxhiu** [comment]
Sorry for the delay everyone. Looking now.

**Comment by michaelhaxhiu** [comment]
Re-posted the job on upwork here:
https://www.upwork.com/jobs/~018c6f2a3bd3e49766

Invited all 3 to the job:
 External issue reporter - gadhiyamanan - $250
 Contributor that fixed the issue - @s77rt - $2000
 Contributor+ that helped on the issue and/or PR - @sobitneupane - $2000

**Comment by michaelhaxhiu** [comment]
I think I made a mistake. It should be $1500 for the C and C+. Not $2000. Fixing that.

**Comment by michaelhaxhiu** [comment]
all paid and correct. Closing
